### ISSUE TITLE
Layered Nav Filter shows an incorrect count when we filter on product attribute pages

### ISSUE THREAD

**Issue by mikkamp** [woocommerce/woocommerce#12272]
1\. When we are on a product attribute page and have selected a second attribute to filter by:
http://woo.dev/pa_color/blue/?filter_size=small

The filter shows all the products with the size small (count = 2) and not the ones with both blue and small.

![Filter](https://cloudup.com/c4Uayzg-LT3+)

Image link: https://cloudup.com/c4Uayzg-LT3

2\. If I go to the shop page and apply both filters:
http://woo.dev/shop/?filter_color=blue&filter_size=small

The filter now shows the products with both size small and color blue (count = 1).

![Filters](https://cloudup.com/cFRF64Xt_Rr+)

Image link: https://cloudup.com/cFRF64Xt_Rr

The layered nav widget uses the function `WC_Query::get_main_tax_query` to get the tax query.
This function in turn gets the tax query from `$wp_the_query->query_vars['tax_query']`

In the first example the tax_query only includes the pa_size = small. It the second example the tax_query includes both pa_color = blue and pa_size = small.

The variable $wp_the_query->tax_query->queries does contain both pa_color = blue and pa_size = small, in both examples.

If I change `function get_main_tax_query` in `class-wc-query.php` from:

```
$tax_query = isset( $args['tax_query'] ) ? $args['tax_query'] : array();
```

to:

```
$tax_query = isset( $wp_the_query->tax_query->queries ) ? $wp_the_query->tax_query->queries : array();
```

When this change is made the counts do show up correctly. Would this change have any other repercussions? Or should the layered nav be the one that handles it differently so it can retrieve the tax query from both the attribute URL and from the filters?

To reproduce this scenario:

- WooCommerce 2.6.7
- Create attribute color (set enable archives), with terms blue, green, red
- Create attribute size, with terms large, medium, small
- Create a product with attributes color = blue and size = small
- Create a second product with attribute size = small

**Comment by mikejolley** [comment]
@mikkamp your links and images are dead.

**Comment by mikkamp** [comment]
@mikejolley The images are on cloudup so they should be loading. I added an extra image link in case they aren't loading correctly.

The URLs are pointing to a dev server. I pasted them there to show the type of URL being used for each example. The first URL is a link to the product attribute page for the color blue, filtering for size=small
The second URL is a link to the shop page filtering both color=blue and size=small.

Let me know if there is any other details needed to reproduce this scenario.
